### ISSUE TITLE
preg_match does not throw an exception

### ISSUE THREAD

**Issue by klaasgeldof** [lorisleiva/laravel-search-string#31]
I don't know why, but in my PHP installation (version 7.4.13) `preg_match` does not throw an exception if the pattern is not valid. The consequence is that the `Rule::regexify` method in this package doesn't work as expected. To get it working, I needed to change this:

```php
protected function regexify($pattern)
{
    try {
        preg_match($pattern, null);
        return $pattern;
    } catch (\Throwable $exception) {
        return '/^' . preg_quote($pattern, '/') . '$/';
    }
}
```

to:

```php
protected function regexify($pattern)
{
    try {
        if (preg_match($pattern, null) === false) {
            throw new \Exception;
        }
        
        return $pattern;
    } catch (\Throwable $exception) {
        return '/^' . preg_quote($pattern, '/') . '$/';
    }
}
```

In the first version, I get a bunch of PHP warnings, but no exception, so the `catch` block is never executed.

I found it strange that I didn't find an issue about this. You can't get correct search results without this change...

**Comment by lorisleiva** [comment]
Hi there 👋 

Thanks for raising this. Indeed, even the official PHP documentation does not mention anything about throwing an exception so I'm not sure why I decided this was the right approach.

I'll add more tests and come up with a more stable solution.

**Comment by lorisleiva** [comment]
Fixed and released in `v1.1.1`. 🚀 

Let me know if that works for you. 🙂

**Comment by klaasgeldof** [comment]
Looks good, thank you Loris!
